lib/reservation.rb

```diff
@@ -0,0 +1,51 @@
+require 'date'
+
+
+module Hotel
+  class Reservation
+
+    attr_reader :ticket_id, :start_date, :end_date
+
+    def initialize(input)
+      @ticket_id = input[:ticket_id]
+      @start_date = input[:start_date]
+      @end_date = input[:end_date]
+      @room = input[:room_id]
+      @reservations = []
+
+      if @end_date < @start_date
+        raise ArgumentError('Invalid Date Range')
+      end
+
+      @reservations << input
+    end
+
+    def self.all
+      return @reservations
```

Comment:
You cannot use instance variables in class methods. Instance variables should generally not exist within class methods because they refer to the overall class.